middleware/proxy/proxy_test.go

```diff
@@ -238,6 +239,118 @@ func TestUnixSocketProxy(t *testing.T) {
 	}
 }
 
+func GetHttpProxy(messageFormat string, prefix string) (*Proxy, *httptest.Server) {
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, messageFormat, r.URL.String())
+	}))
+
+	ts.Start()
```

Comment:
Since you're starting the server right away, you might be able to just use `httptest.NewServer()`